hack/generator/pkg/astmodel/armconversion/from_arm_conversion_function.go

```diff
@@ -0,0 +1,431 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package armconversion
+
+import (
+	"fmt"
+	"github.com/Azure/k8s-infra/hack/generator/pkg/astbuilder"
+	"github.com/Azure/k8s-infra/hack/generator/pkg/astmodel"
+	"go/ast"
+	"go/token"
+)
+
+func (c *ArmConversionFunction) asConvertFromArmFunc(
+	codeGenerationContext *astmodel.CodeGenerationContext,
+	receiver astmodel.TypeName,
+	methodName string) *ast.FuncDecl {
+
+	// Determine the type we're operating on
+	kubeType := getReceiverObjectType(codeGenerationContext, receiver)
+
+	var fromArmStatements []ast.Stmt
+
+	receiverIdent := ast.NewIdent(c.idFactory.CreateIdentifier(receiver.Name(), astmodel.NotExported))
+
+	// Note: If we have a property with these names we will have a compilation issue in the generated
+	// code. Right now that doesn't seem to be the case anywhere but if it does happen we may need
+	// to harden this logic some to choose an unused name.
+	inputIdent := ast.NewIdent("armInput")
+	typedInputIdent := ast.NewIdent("typedInput")
+
+	// perform a type assert
+	fromArmStatements = append(fromArmStatements, &ast.AssignStmt{
+		Lhs: []ast.Expr{
+			typedInputIdent,
+			ast.NewIdent("ok"),
+		},
+		Tok: token.DEFINE,
+		Rhs: []ast.Expr{
+			&ast.TypeAssertExpr{
+				X:    inputIdent,
+				Type: ast.NewIdent(c.armTypeName.Name()),
+			},
+		},
+	})
```

Comment:
I think this is fixed but a bit hard to tell because I refactored quite a bit and GitHub isn't letting me easily look at the context here -- going to resolve this and let me know if you think there are some places where another helper would be useful.